uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/apis/work-pattern-api.js

```diff
@@ -13,7 +13,43 @@ define([
     return api.extend({
 
       /**
-       * This method returns the calendar for the given contact(s) and period,
+       * Assigns a work pattern to a contact
+       *
+       * @param {string} contactId
+       * @param {string} workPatternID
+       * @param {string} effectiveDate
+       * @param {string} effectiveEndDate
+       * @param {string} changeReason
+       * @param {object} params - additional parameters
+       * @return {Promise} Resolved with {Array} All Work Patterns
+       */
+      assignWorkPattern: function (contactId, workPatternID, effectiveDate, effectiveEndDate, changeReason, params) {
```

Comment:
@deb1990 i think in the end it's a matter of preferences. I personally like just one `param` parameter because when you call the method everything is very explicit and simple to understand, and also because you don't need to have a separate param for the "other" params (the optional ones)
```js
assignWorkPattern({
  contactId: 2,
  workPatternId: 3,
  effectiveDate: '2017-01-01',
  endDate: '2017-02-02',
  changeReason: 'foobar',
  someAdditionalParam: 'barbaz'
});
```
compare to
```js
assignWorkPattern(2, 3, '2017-01-01', '2017-02-02', 'foobar',  { someAdditionalParam: 'barbaz' });
```
in the second example, you have to remember exactly what each parameter means in the method's signature, which is not convenient nor expressive with methods with long signatures


Anyway although i prefer the first version, I'm fine with both. Just my 2 cents